mixer/pkg/protobuf/yaml/dynamic/builder.go

```diff
@@ -0,0 +1,280 @@
+		}
+
+		switch fd.GetType() {
+		case descriptor.FieldDescriptorProto_TYPE_STRING:
+			var ma []interface{}
+			if fd.IsRepeated() {
+				if ma, ok = v.([]interface{}); !ok {
+					return nil, fmt.Errorf("unable to process: %v, got %T, want: []interface{}", fd, v)
+				}
+			} else {
+				ma = []interface{}{v}
+			}
+
+			for _, vv := range ma {
+				var fld *field
+				if fld, err = c.buildPrimitiveField(vv, fd, isMap && k == "key"); err != nil {
+					return nil, fmt.Errorf("unable to encode field: %v. %v", fd, err)
+				}
+				me.fields = append(me.fields, fld)
+			}
+
+			// primitives
+		case descriptor.FieldDescriptorProto_TYPE_DOUBLE,
+			descriptor.FieldDescriptorProto_TYPE_FLOAT,
+			descriptor.FieldDescriptorProto_TYPE_INT64,
+			descriptor.FieldDescriptorProto_TYPE_UINT64,
+			descriptor.FieldDescriptorProto_TYPE_INT32,
+			descriptor.FieldDescriptorProto_TYPE_FIXED64,
+			descriptor.FieldDescriptorProto_TYPE_FIXED32,
+			descriptor.FieldDescriptorProto_TYPE_BOOL,
+			descriptor.FieldDescriptorProto_TYPE_UINT32,
+			descriptor.FieldDescriptorProto_TYPE_SFIXED32,
+			descriptor.FieldDescriptorProto_TYPE_SFIXED64,
+			descriptor.FieldDescriptorProto_TYPE_SINT32,
+			descriptor.FieldDescriptorProto_TYPE_SINT64,
+			descriptor.FieldDescriptorProto_TYPE_ENUM:
+
+			packed := fd.IsPacked() || fd.IsPacked3()
+			if fd.IsRepeated() && !packed {
+				return nil, fmt.Errorf("unpacked primitives not supported: %v", fd)
+			}
+
+			var fld *field
+			if fld, err = c.buildPrimitiveField(v, fd, isMap && k == "key"); err != nil {
+				return nil, fmt.Errorf("unable to encode field: %v. %v", fd, err)
+			}
+			me.fields = append(me.fields, fld)
+		case descriptor.FieldDescriptorProto_TYPE_MESSAGE:
+			m := c.resolver.ResolveMessage(fd.GetTypeName())
+			if m == nil {
+				return nil, fmt.Errorf("unable to resolve message '%s'", fd.GetTypeName())
+			}
+
+			var ma []interface{}
+			if m.GetOptions().GetMapEntry() { // this is a Map
+				ma, err = convertMapToMapentry(v)
+				if err != nil {
+					return nil, fmt.Errorf("unable to process: %v, %v", fd, err)
+				}
+			} else if fd.IsRepeated() { // map entry is always repeated.
+				ma, ok = v.([]interface{})
+				if !ok {
+					return nil, fmt.Errorf("unable to process: %v, got %T, want: []interface{}", fd, v)
+				}
+			} else {
+				ma = []interface{}{v}
+			}
+
+			// now maps, messages and repeated messages all look the same.
+			if err = c.buildMessageField(ma, m, fd, &me); err != nil {
+				return nil, err
+			}
+
+		default:
+			return nil, fmt.Errorf("field not supported '%v'", fd)
+		}
+	}
+
+	// sorting fields is recommended
+	sort.Slice(me.fields, func(i, j int) bool {
+		return me.fields[i].number < me.fields[j].number
+	})
+
+	return me, nil
+}
+
+// buildPrimitiveField handles encoding of single and repeated packed primitives
+func (c Builder) buildPrimitiveField(v interface{}, fd *descriptor.FieldDescriptorProto,
+	static bool) (*field, error) {
```

Comment:
`static bool` field is It is unclear to me, since this func is called with values `isMap && k == "key"`. Consider renaming or comment

Reply:
updated to `noExpr`